src/storage/mod.rs

```diff
@@ -0,0 +1,606 @@
+    >(
+        tx: Sender<StorageChannelPayload>,
+        key: S::Key,
+    ) -> Self {
+        let getter = move || {
+            let data = S::get::<T>(&key).unwrap();
+            StorageChannelPayload::new(data)
+        };
+        Self {
+            getter: Box::new(getter),
+            tx: Arc::new(tx),
+        }
+    }
+
+    /// Gets the latest data from storage and sends it to the channel.
+    pub fn get_and_send(&self) -> Result<(), SendError<StorageChannelPayload>> {
+        let payload = (self.getter)();
+        self.tx.send(payload)
+    }
+}
+
+// End StorageSenderEntry
+
+// Start StorageChannelPayload
+
+/// A payload for a storage channel that contains the latest value from storage.
+#[derive(Clone, Debug)]
+pub struct StorageChannelPayload {
+    data: Arc<dyn Any + Send + Sync>,
+}
+
+impl StorageChannelPayload {
+    /// Creates a new StorageChannelPayload
+    pub fn new<T: Send + Sync + 'static>(data: T) -> Self {
+        Self {
+            data: Arc::new(data),
+        }
+    }
+
+    /// Gets the data from the payload
+    pub fn data<T: 'static>(&self) -> Option<&T> {
+        self.data.downcast_ref::<T>()
+    }
+}
+
+impl Default for StorageChannelPayload {
+    fn default() -> Self {
+        Self { data: Arc::new(()) }
+    }
+}
+// End StorageChannelPayload
+
+// Start helper functions
+
+/// Serializes a value to a string and compresses it.
+pub(crate) fn serde_to_string<T: Serialize>(value: &T) -> String {
+    let serialized = to_allocvec(value).unwrap();
+    let compressed = yazi::compress(
+        &serialized,
+        yazi::Format::Zlib,
+        yazi::CompressionLevel::BestSize,
+    )
+    .unwrap();
+    let as_str: String = compressed
+        .iter()
+        .flat_map(|u| {
+            [
+                char::from_digit(((*u & 0xF0) >> 4).into(), 16).unwrap(),
+                char::from_digit((*u & 0x0F).into(), 16).unwrap(),
+            ]
+            .into_iter()
+        })
+        .collect();
+    as_str
+}
+
+#[allow(unused)]
+/// Deserializes a value from a string and unwraps errors.
+pub(crate) fn serde_from_string<T: DeserializeOwned>(value: &str) -> T {
+    try_serde_from_string(value).unwrap()
+}
+
+/// Deserializes and decompresses a value from a string and returns None if there is an error.
+pub(crate) fn try_serde_from_string<T: DeserializeOwned>(value: &str) -> Option<T> {
+    let mut bytes: Vec<u8> = Vec::new();
+    let mut chars = value.chars();
+    while let Some(c) = chars.next() {
+        let n1 = c.to_digit(16)?;
+        let c2 = chars.next()?;
+        let n2 = c2.to_digit(16)?;
+        bytes.push((n1 * 16 + n2) as u8);
+    }
+    match yazi::decompress(&bytes, yazi::Format::Zlib) {
+        Ok((decompressed, _)) => match postcard::from_bytes(&decompressed) {
+            Ok(v) => Some(v),
+            Err(err) => {
+                tracing::error!("Error deserializing value from storage: {:?}", err);
```

Comment:
Trying out the latest branch, I get:

```
   --> /Users/srid/.cargo/git/checkouts/dioxus-std-d24bc7446db44106/72d2346/src/storage/mod.rs:601:13
    |
601 |             tracing::error!("Error decompressing value from storage: {:?}", err);
    |             ^^^^^^^ use of undeclared crate or module `tracing`
```

I see that there is no `tracing` dependency in this project `Cargo.toml`, so it is rather stranger that CI still passes.

Reply:
Need to add `storage` feature in Github Actions? cf. 

https://github.com/DioxusLabs/dioxus-std/blob/38dfc96b145635b67af5391f19f86838a77a46a9/.github/workflows/testing.yml#L32-L33

Reply:
Fixed, it looks like clippy wasn't enabled for any features as well